editor/components/post-taxonomies/flat-term-selector.js

```diff
@@ -104,27 +104,27 @@ class FlatTermSelector extends Component {
 	}
 
 	findOrCreateTerm( termName ) {
-		return new Promise( ( resolve, reject ) => {
-			// Tries to create a term or fetch it if it already exists
-			const basePath = wp.api.getTaxonomyRoute( this.props.slug );
-			apiRequest( {
-				path: `/wp/v2/${ basePath }`,
-				method: 'POST',
-				data: { name: termName },
-			} ).then( resolve, ( xhr ) => {
-				const errorCode = xhr.responseJSON && xhr.responseJSON.code;
+		const basePath = wp.api.getTaxonomyRoute( this.props.slug );
+		// Tries to create a term or fetch it if it already exists
+		return apiFetch( {
+			path: `/wp/v2/${ basePath }`,
+			method: 'POST',
+			data: { name: termName },
+		} ).catch( ( response ) =>
+			response.json().then( ( body ) => {
```

Comment:
So in `.then` we receive a JSON payload by default ? But `.catch` we receive `response` always ? And in `.then` we have the option to receive `response` by passing `parse: false` ? Feels like an inconsistent developer experience.

Reply:
Yes, what's the alternative you suggest? Try to parse the response in errors as well?

Reply:
That, or just abandon the auto-parse convenience. Consistency could have the bigger benefit.

Reply:
I personally prefer to keep the auto-parsing because of the duplication it can create. I'm fine with consistency between `then/catch`

Reply:
I think parsing error response bodies makes sense since this package is specifically designed to work with the WP REST API, which we know returns nicely formatted error responses:

```json
{
    "code": "rest_forbidden",
    "data": {
        "status": 401
    },
    "message": "Sorry, you are not allowed to do that."
}
```

Most error handling can then be as simple as:

```js
apiFetch( ... ).catch( ( { message } ) => {
	console.error( message );
} );
```

If we cannot parse the error response body (e.g. because of a connection issue), then we should throw a similarly shaped object, e.g.:

```js
{
    code: "unknown_error",
    message: "An unknown error occurred.",
}
```